lib/msf/core/exploit/remote/http_client.rb

```diff
@@ -285,6 +263,42 @@ def connect_ws(opts={}, timeout = 20)
     socket = http_client.conn
     socket.extend(Rex::Proto::Http::WebSocket::Interface)
   end
+  
+  # Defines a proc to log HTTP requests and responses
+  # 
+  # @param http_trace [Bool] A Boolean representing the datastore['HttpTrace']
+  # to check if HttpTrace is set or unset.
+  #
+  # @return [Proc] A Proc object to log HTTP requests and responses, or nil if datastore['HttpTrace'] is unset
+  def set_http_trace_proc(http_trace)
+    proc_httptrace = nil
+    if http_trace
+      proc_httptrace = proc { |request, response| 
+        request_color, response_color = 
+          (datastore['HttpTraceColors'] || 'red/blu').split('/').map { |color| "%bld%#{color}" }
+        
+        request = request.to_s(headers_only: datastore['HttpTraceHeadersOnly'])
```

Comment:
Why is this being done here and not in your [lib/metasploit/framework/login_scanner/http.rb](https://github.com/3V3RYONE/metasploit-framework/pull/7/files#diff-b00cbc1b487d443e3b3fe256e2489c8fe2c98ddd5d84a5c05d31bb9cb4f64633) definition?

Reply:
Also on a similar note shouldn't this be calling `to_terminal_output`? Why is `.to_s` being used here?

Reply:
https://github.com/rapid7/metasploit-framework/search?q=to_terminal_output shows that only the packet response has `to_terminal_output` call on it to convert from a packet to proper terminal output. Therefore these changes are in line with that current code pattern.